robot_controllers/src/diff_drive_base.cpp

```diff
@@ -321,16 +321,19 @@ void DiffDriveBaseController::update(const ros::Time& now, const ros::Duration&
   // Lock mutex before updating
   boost::mutex::scoped_lock lock(odom_mutex_);
 
-  // Update stored odometry pose...
-  theta_ += th/2.0;
-  odom_.pose.pose.position.x += d*cos(theta_);
-  odom_.pose.pose.position.y += d*sin(theta_);
-  theta_ += th/2.0;
-  odom_.pose.pose.orientation.z = sin(theta_/2.0);
-  odom_.pose.pose.orientation.w = cos(theta_/2.0);
-  // ...and twist
-  odom_.twist.twist.linear.x = dx;
-  odom_.twist.twist.angular.z = dr;
+  if (std::isfinite(left_vel) && std::isfinite(right_vel))
+  {
+    // Update stored odometry pose...
+    theta_ += th/2.0;
+    odom_.pose.pose.position.x += d*cos(theta_);
+    odom_.pose.pose.position.y += d*sin(theta_);
+    theta_ += th/2.0;
+    odom_.pose.pose.orientation.z = sin(theta_/2.0);
+    odom_.pose.pose.orientation.w = cos(theta_/2.0);
+    // ...and twist
+    odom_.twist.twist.linear.x = dx;
+    odom_.twist.twist.angular.z = dr;
+  }
```

Comment:
How do you feel about a throttled log message with values left_vel, right_vel for else statement?